File: Coding-Interviews/03-数组中重复的数字.cpp
#include <vector>
using namespace std;
class Solution
{
public:
    int findRepeatNumber(vector<int>& nums)
    {
        vector<int> hash(nums.size(), 0);
        for (auto const i : nums) {
            if (hash[i] != 0)
                return i;
            hash[i]++;
        }
        return -1;
    }
};

File: Coding-Interviews/03-数组中重复的数字.py
class Solution:
    def findRepeatNumber(self, nums: [int]) -> int:
        i = 0
        while i < len(nums):
            if (nums[i] == i):
                i += 1
                continue
            else:
                if nums[i] == nums[nums[i]]:
                    return nums[i]
                nums[nums[i]], nums[i] = nums[i], nums[nums[i]]
        return -1


if __name__ == "__main__":
    Solution().findRepeatNumber([1, 2, 3, 3, 4])
